### ISSUE TITLE
Possible bug on calculating parent nodes in differentiable astar module on large images.

### ISSUE THREAD

**Issue by SamuelMastrelli** [omron-sinicx/neural-astar#27]
As the title says, when the differentiable astar module is fed the type of images below, it seems it cant calculate the whole path. In this case I i used VanillaAstar with the differentiable astar module:
![image](https://github.com/user-attachments/assets/bdca25bc-dc81-476c-92d7-b4d83d5b748c)

It reaches the goal from the start during the exploration, but i doesn't complete the path.

**Comment by yonetaniryo** [comment]
Thank you for the question! Can you provide the complete information to reproduce your issue?

**Comment by SamuelMastrelli** [comment]
Certainly! What i'm trying to do is feeding the algorithm this type of images
![image](https://github.com/user-attachments/assets/8194a00d-779a-44eb-8c88-9d9308152aed)
that represent closed enviroments like buildings, houses ecc. 
After choosing start and goal inside the contours, I use the 4 dim tensor which reprensents the map_design, the tensor which represents the star points and the tensor representing the goal points as input for VanillaAstar using the differentiable astar module. 
Most of the time, the result is that when reconstructing the path, it stops and doesn't connect the goal with the start. 
This cases seems to have something in common: a single piece of the path is not connected.
VanillaAstar with pqstar works fine.

**Comment by SamuelMastrelli** [comment]
I probably just realized what is the problem. I convert jpg images to tensor with the following:


```
        img = cv2.imread(dir + "/" + cluster + "/" + image)
        print(image)

        # Converti l'immagine in scala di grigi
        gray_image = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)

        # Creare una maschera binaria delle aree bianche
        _, binary_mask = cv2.threshold(gray_image, 240, 255, cv2.THRESH_BINARY)

        # Invertire la maschera
        binary_mask_inv = cv2.bitwise_not(binary_mask)

        # Trova i contorni dell'edificio
        contours, _ = cv2.findContours(binary_mask_inv, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)

        # Creare una maschera di riempimento
        fill_mask = np.zeros_like(gray_image)

        # Riempire solo le aree esterne all'edificio
        cv2.drawContours(fill_mask, contours, -1, (255), thickness=cv2.FILLED)

        # Invertire la maschera di riempimento
        #fill_mask_inv = cv2.bitwise_not(fill_mask)

        # Applicare la maschera per cambiare le aree esterne bianche in nero
        #img[fill_mask_inv == 255] = [0, 0, 0]

        transform = transforms.Compose([
                    transforms.ToTensor()
                ])

        image_tensor = transform(img)


        map_design = torch.clamp(image_tensor.mean(0), 0, 1)
```
I tried now with pngs and it works fine. Thanks anyway. Sorry for bothering you

**Comment by SamuelMastrelli** [comment]
Image conversion error
